model/flow/quorum_certificate.go

```diff
@@ -14,7 +14,7 @@ type QuorumCertificate struct {
 	// In order to distinguish the signature types, the SigData has to be deserialized. Specifically,
 	// the field `SigData.SigType` (bit vector) indicates for each signer which sig type they provided.
 	// For collection cluster, the SignerIDs includes all the staking sig signers.
-	SignerIDs []Identifier
+	SignerIndices []byte
```

Comment:
Q: Why changing the field here?

A: I was evaluating two options:
1. Keeping the QC's `SignerIDs` field unchanged, and decode the signer indices from the header and query the signer IDs when creating the QC.
2. Change the field here

Option 1 seems better since it can keep the QC's structure unchanged, however it makes a lot of things harder, for instance, the [`BlockFromFlow`](https://github.com/onflow/flow-go/blob/master/consensus/hotstuff/model/block.go#L21) function is a pure function that can convert a block header into a hotstuff Block. Since header uses signerIndices but QC uses signer IDs, then we need to query signerID by signerIndices first. And that query is stateful, and would have to be mocked in tests, it would be more complex to do. 

Instead, I picked Option 2. Basically, we are delaying the query to convert signer indices into signer IDs until its needed by certain modules. And since some modules can work directly with signer indices, keeping it as is could skip some back-and-force conversions.